### PR TITLE
26 implement kafka consumer and producer for crawler

### LINKED ISSUE #26: Implement Kafka consumer and producer for Crawler
LFI: URL's with query parameters.
XSS: URL's with query parameters and forms.
SQLi:  URL's with forms.
BA: URL's with login form.

### PR DESCRIPTION
**Pull request by DmytroTHR**

### CODE REVIEW

File: cmd/app/main.go [hunk 1]
```diff
@@ -2,78 +2,162 @@ package main
...
+	if err != nil {
+		log.Panicln("Error loading .env file: ", err)
+	}
 }
 
 func main() {
-	//receive input - site to crawl
-	//TODO: unmock site name receiving, when get input data format
-	providedURL, err := url.Parse(mockSiteName)
+	app := new(Config)
```
Comment (avobl) [inline cmd/app/main.go:35]: You can use chain here, like: `app := new(Config).initPubSub()`

File: cmd/app/config.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	var result [crawler.NumOfBodyParams]bool
+	result[crawler.HasFormTag] = filters[crawler.HasFormTag]
+	result[crawler.HasQueryParameter] = filters[crawler.HasQueryParameter]
+
+	return result
+}
+
+func EnvVarOfType(varName string, varType int) any {
```
Comment (avobl) [inline cmd/app/config.go:41]: // Do you really need this function? I don't see any benefit. It doesn't make your code shorter, although brings some complexity.

**Review by avobl — APPROVED** [review]

File: cmd/app/main.go (hunk 1 above)
Comment (avobl) [inline cmd/app/main.go:35]: You can use chain here, like : `app := new(Config).initPubSub()`

**Review by avobl — APPROVED** [review]

File: cmd/app/main.go (hunk 1 above)
Comment (DmytroTHR) [reply cmd/app/main.go:35]: Don't think so. To use it in a chain as in a Builder pattern all parts of the chain should return the type of the struct I want to create. This one is not the case I think.

Comment (DmytroTHR) [reply cmd/app/main.go:35]: This comment was duplicated somehow and should be deleted I suppose. Replied to this comment above.

File: cmd/app/config.go (hunk 2 above)
Comment (DmytroTHR) [reply cmd/app/config.go:41]: I'm not sure if I need it. I use it to identically receive ENV parameters in the code (whether I need them just as a string, an int, or some other more complicated type). Maybe there's a better way of doing so, but I didn't find it.

**Review by away4people — APPROVED** [review]

File: cmd/app/config.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	"LFI-check":  fillTestFilter("LFI", false, true),
+}
+
+func fillTestFilter(testName string, filters ...bool) [crawler.NumOfBodyParams]bool {
+	if len(filters) != crawler.NumOfBodyParams {
+		log.Panicln("Wrong filters number for test", testName)
+	}
+
+	var result [crawler.NumOfBodyParams]bool
+	result[crawler.HasFormTag] = filters[crawler.HasFormTag]
```
Comment (PavelAD) [inline cmd/app/config.go:34]: You will recive panic here

Comment (DmytroTHR) [reply cmd/app/config.go:34]: Why should I if it's an array of a fixed size?
Not receiving it now.

Comment (PavelAD) [reply cmd/app/config.go:34]: Should use make for create map
If define map using var and try to set up something you receive the panic, for example: https://go.dev/play/p/ar6G6wvchfu

Comment (DmytroTHR) [reply cmd/app/config.go:34]: Hmm, ok. But again, it's not a map. It is an array of a fixed size. And it works pretty well now.

Comment (PavelAD) [reply cmd/app/config.go:34]: Ahh, yes you are right 
Sorry, my bad 
I did not see classic array for a long time)

**Pull request merged**